server/app/repository/ApiKeyRepository.java

```diff
@@ -0,0 +1,54 @@
+package repository;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+
+import io.ebean.DB;
+import io.ebean.Database;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.ApiKey;
+
+/** Provides an asynchronous API for persistence and query of {@link ApiKey} instances. */
+public class ApiKeyRepository {
+  private final Database database;
+  private final DatabaseExecutionContext executionContext;
+
+  @Inject
+  public ApiKeyRepository(DatabaseExecutionContext executionContext) {
+    this.database = DB.getDefault();
+    this.executionContext = checkNotNull(executionContext);
+  }
+
+  /** Insert a new ApiKey record using a thread from {@link DatabaseExecutionContext}. */
```

Comment:
Suggestion: Replace DatabaseExecutionContext with more general "async" language.  The public API here doesn't really discuss the threading, if anything add DatabaseExecutionContext to the class comment as it underlies the general operation.

Reply:
done